internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/writer/VeniceWriterTest.java

```diff
@@ -227,4 +230,53 @@ public void testVeniceWriterClose(boolean doFlush) {
       }
     }
   }
+
+  /**
+   * This is a regression test for the VeniceWriter issue where the VeniceWriter could run into
```

Comment:
Thanks for writing the integration test. That is good, but is it possible to also add a unit test which repros this issue?